src/vowpal_wabbit_next/_core.pyi

```diff
@@ -3,20 +3,67 @@ import vowpal_wabbit_next._core
 import typing
 
 __all__ = [
+    "CBLabel",
     "DenseParameters",
     "Example",
     "LabelType",
     "ModelDelta",
+    "MulticlassLabel",
     "PredictionType",
+    "SimpleLabel",
     "Workspace"
 ]
 
 
+class CBLabel():
+    def __init__(self, *, label: typing.Optional[typing.Union[typing.Tuple[float, float], typing.Tuple[int, float, float]]] = None, weight: float = 1.0, shared: bool = False) -> None: 
+        """
+        A label representing a contextual bandit problem.
+
+        Args:
+          label (Optional[Union[Tuple[float, float], Tuple[int, float, float]]): This is (action, cost, probability). The same rules as VW apply for if the action can be left out of the tuple.
+          weight (float): The weight of the example.
+          shared (bool): Whether the example is shared. This is only used for ADF examples and must be the first example. There can only be one shared example per ADF example list.
+        """
+    @property
+    def label(self) -> typing.Optional[typing.Union[typing.Tuple[float, float], typing.Tuple[int, float, float]]]:
+        """
+            The label for the example. The format of the label is (action, cost, probability).
```

Comment:
what is Tuple[float, float] option?

Reply:
When the action is not relevant (like in vw adf format), its a messy special case in VW. I was trying to smooth over it but it is confusing